### DISCUSSION TITLE
collection size and filters

### DISCUSSION THREAD

**Discussion by tlyfoung-widen** [lariat-js/playwright#36] [Ideas]
Could we add .length and ways to filter elements of a collection?

**Comment by mskelton** [comment]
@tlyfoung-widen Could you explain what you mean by filtering?  Locators can be filtered using [`Locator.filter()`](https://playwright.dev/docs/api/class-locator#locator-filter), and `Collection.el` supports `has` and `hasText` already.

If you want to get a count, you can do that using the [`Locator.count()`](https://playwright.dev/docs/api/class-locator#locator-count) method, though generally using the `toHaveCount` matcher is preferred.

**Reply by tlyfoung-widen** [reply]
I found out that I need to use the collection element's root to be able to use `toHaveCount`. I was also hoping to filter a collection by status (for product task, uncomplete/complete) and count the number of the tasks. I ended up not using count since the elements are present in the page, but not displayed. Useful learning though. Thank you
